app/main.py

```diff
@@ -1,8 +1,46 @@
 class Car:
-    # write your code here
-    pass
+    def __init__(self,
+                 comfort_class: int,
+                 clean_mark: int,
+                 brand: str) -> None:
+        self.comfort_class = comfort_class
+        self.clean_mark = clean_mark
+        self.brand = brand
 
 
 class CarWashStation:
-    # write your code here
-    pass
+    def __init__(self,
+                 distance_from_city_center: float,
+                 clean_power: int,
+                 average_rating: float,
+                 count_of_ratings: int) -> None:
+        self.distance_from_city_center = distance_from_city_center
+        self.clean_power = clean_power
+        self.average_rating = round(average_rating, 1)
+        self.count_of_ratings = count_of_ratings
+
+    def serve_cars(self, cars: list[Car]) -> int:
+        income = 0
+        for car in cars:
+            if car.clean_mark < self.clean_power:
+                income += self.calculate_washing_price(car)
+            self.wash_single_car(car)
+        return round(income, 1)
+
+    def calculate_washing_price(self, car: Car) -> float:
+        return round(
+            (car.comfort_class * (self.clean_power - car.clean_mark)
+             * self.average_rating / self.distance_from_city_center),
+            1
+        )
+
+    def wash_single_car(self, car: Car) -> None:
+        if self.clean_power > car.clean_mark:
+            car.clean_mark = self.clean_power
+
+    def rate_service(self, new_rating: int) -> None:
+        self.count_of_ratings += 1
+        self.average_rating = round(
+            ((self.average_rating * (self.count_of_ratings - 1))
+             + new_rating) / self.count_of_ratings, 1
+        )
```

Comment:
Code Style: Again, as per the first point in the checklist, this long mathematical expression should be split into additional variables or broken after binary operations. This will make the code more readable and easier to understand.